### DISCUSSION TITLE
Requested features from GlitchSoc

### DISCUSSION THREAD

**Discussion by marrus-sh** [cat-KIBI-glitch/mastodon#24] [Request for Comments]
The following features were requested for GlitchSoc but never implemented :—

+ Out‐of‐body hashtags: glitch-soc/mastodon#5
+ Disabling the local or federated timelines for an instance
+ Thread collapsing (this is imagined as a frontend feature): glitch-soc/mastodon#28
+ Decrease post visibility on federation: glitch-soc/mastodon#79
+ Notification merging: glitch-soc/mastodon#130
+ Card prefetching: glitch-soc/mastodon#187
+ Load about pages from a file: glitch-soc/mastodon#251
+ Reactions: glitch-soc/mastodon#323
+ Flavour‐specific configurations: glitch-soc/mastodon#350
+ Limiting number of followed accounts; providing time estimates: glitch-soc/mastodon#597

Are we interested in any of these?

**Comment by aescling** [comment]
my thoughts, which i guess TL;DRs to “i like most of these ideas” lol

## wants

* monsterfork actually did out of body hashtags and i always thought it was a good idea
* i’ve personally always wanted to be able to disable the public timeline (there was a time when i thought that if i was gonna admin an instance, i would hack away support for the public timeline lol). if i were going to run a a single user instance (not that i would) i’d definitely make use of that
* thread collapsing would be a great feature imo

## things i think are worth supporting

* notification merging is a good idea imo
* i feel like card prefetching is a must have feature for post previews, which is the feature i would actually suggest here
* loading about pages from a file is a good idea a think
* i think the fire demon was working on reactions for monsterfork but never actually finished it; personally i was never big on the idea but it like, makes sense. generic favourite is probably actually the worse idea though so
* flavour-specific configuration makes sense to me
* time estimates and/or post frequency data both seem worth making available to users to me

## questions

* why would you want decreased last visibility on federation? i think it’s a neat idea though

**Reply by marrus-sh** [reply]
Mainly the idea behind decreased visibility on federation is to have federated posts which don’t show up in public timelines elsewhere, but show up in public timelines locally (`public` → `unlisted`), or posts which can be boosted by local users but not by just anyone (`public`/`unlisted` → `followers-only`).

**Reply by aescling** [reply]
i feel like monsterfork actually did that one too at some point now that i think about it

anyway regardless i can see how that could be desirable and i would be up for implementing it

(i am getting very worried by the number of features i would like to see added lol)

**Reply by marrus-sh** [reply]
wow imagine if over the past half‐decade eugen had implemented any of this!! 🙀

(i jest but the list used to be even longer and some of the requests to glitch-soc actually *have* been implemented in upstream now.)

**Reply by marrus-sh** [reply]
for real though the answer to this worry is to recruit developers and work in solidarity with other forks / instances lmao 😂

**Comment by marrus-sh** [comment]
I could have sworn this got merged but I guess it never did? https://github.com/glitch-soc/mastodon/pull/257

It’ll need to be re·implemented now, probably.

**Reply by aescling** [reply]
oh i *like* that
